### PR TITLE
Insert from select

### PR DESCRIPTION
**Pull request by 05storm26**
Fixes: #231

### CODE REVIEW

File: src/query/insert.rs [hunk 1]
```diff
@@ -3,6 +3,16 @@ use crate::{
...
 
+/// Represents a value source that can be used in an insert query.
+///
+/// [`InsertValueSource`] is a node in the expression tree and can represent a raw value set
+/// ('VALUES') or a select query.
+#[derive(Debug, Clone)]
+pub enum InsertValueSource {
+    VALUES(Vec<Vec<SimpleExpr>>),
+    QUERY(SelectStatement),
+}
```
Comment (billy1624) [inline src/query/insert.rs:14]: I think it's better to `Box` the `SelectStatement`, i.e. `Box<SelectStatement>`.
CC @tyt2y3

Comment (05storm26) [reply src/query/insert.rs]: Done

File: src/query/insert.rs [hunk 2]
```diff
@@ -3,6 +3,16 @@ use crate::{
...
 
+/// Represents a value source that can be used in an insert query.
+///
+/// [`InsertValueSource`] is a node in the expression tree and can represent a raw value set
+/// ('VALUES') or a select query.
+#[derive(Debug, Clone)]
+pub enum InsertValueSource {
+    VALUES(Vec<Vec<SimpleExpr>>),
+    QUERY(Box<SelectStatement>),
+}
```
Comment (billy1624) [inline src/query/insert.rs:14]: Could we rename variants as `Values` and `Select`?

Comment (05storm26) [reply src/query/insert.rs]: Done

**Review by billy1624 — APPROVED** [review]
Ty!

**Review by tyt2y3 — APPROVED** [review]
Wow. I absolutely love the API

**Comment by tyt2y3** [comment]
okay. one question. why use case does pub-ing `value_source` allows?

**Comment by 05storm26** [comment]
@tyt2y3 I don't know. Should we make it non public? Should I open a new PR about it?

**Pull request merged**